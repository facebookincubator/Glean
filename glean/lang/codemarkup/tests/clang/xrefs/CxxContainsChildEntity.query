query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
