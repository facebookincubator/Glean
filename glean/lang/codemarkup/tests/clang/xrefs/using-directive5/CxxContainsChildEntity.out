[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N1" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "T" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N1" } } }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 11,
                "columnEnd": 11
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N2" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 17,
                "columnBegin": 1,
                "lineEnd": 27,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N21" },
                  "parent": { "key": { "name": { "key": "N2" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 18,
                "columnBegin": 1,
                "lineEnd": 26,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N21" },
                  "parent": { "key": { "name": { "key": "N2" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 18,
                "columnBegin": 1,
                "lineEnd": 26,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N21" },
                        "parent": { "key": { "name": { "key": "N2" } } }
                      }
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 22,
                "columnBegin": 1,
                "lineEnd": 24,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  }
]